src/Sarif.TestUtilities/FileDiffingTests.cs

```diff
@@ -53,37 +48,38 @@ public FileDiffingTests(ITestOutputHelper outputHelper)
 
         protected virtual string TestLogResourceNameRoot => "Microsoft.CodeAnalysis.Sarif.UnitTests.TestData." + TypeUnderTest;
 
-        protected virtual string ConstructTestOutputFromInputResource(string inputResource) { return string.Empty; }
+        protected virtual string ConstructTestOutputFromInputResource(string inputResourceName) { return string.Empty; }
 
         protected virtual bool RebaselineExpectedResults => false;
 
-        protected virtual void RunTest(string resourceName)
+        protected virtual void RunTest(string inputResourceName, string expectedOutputResourceName = null)
```

Comment:
>RunTest [](start = 31, length = 7)

Here's how we handle one-to-many tests. Create an individual test referencing the common input, but generate an alternate output name. See the optional data visitor tests for a specific example.

Reply:
So if an operation (such as a JSchema code gen from a schema) generates multiple outputs, we actually have to run the operation as many times as there are output files, even though all the outputs are generated each time? I'm not wild about that.

---
In reply to: [247382063](https://github.com/Microsoft/sarif-sdk/pull/1212#discussion_r247382063) [](ancestors = 247382063)

Reply:
glad to discuss further.

---
In reply to: [247659167](https://github.com/Microsoft/sarif-sdk/pull/1212#discussion_r247659167) [](ancestors = 247659167,247382063)